rholang/src/main/bnfc/rholang_mercury.cf

```diff
@@ -61,18 +61,38 @@ PAnd.            Proc5  ::= Proc5 "and" Proc6 ;
 POr.             Proc4  ::= Proc4 "or" Proc5 ;
 PSend.           Proc3  ::= Name Send "(" [Proc] ")" ;
 PContr.          Proc2  ::= "contract" Name "(" [Name] NameRemainder")" "=" "{" Proc "}" ;
-PInput.          Proc2  ::= "for" "(" Receipt ")" "{" Proc "}" ;
+PInput.          Proc2  ::= "for" "(" [Receipt] ")" "{" Proc "}" ;
 PChoice.         Proc2  ::= "select" "{" [Branch] "}" ;
 PMatch.          Proc2  ::= "match" Proc4 "{" [Case] "}" ;
 PBundle.         Proc2  ::= Bundle "{" Proc "}" ;
+PLet.            Proc2  ::= "let" Decl Decls "in" "{" Proc "}" ;
 PIf.             Proc1  ::= "if" "(" Proc ")" Proc2 ;
 -- Use precedence to force braces around an interior if.
 PIfElse.      Proc1 ::= "if" "(" Proc ")" Proc2 "else" Proc1 ;
 PNew.         Proc1 ::= "new" [NameDecl] "in" Proc1 ;
+PSendSynch.   Proc1 ::= Name "!?" "(" [Proc] ")" SynchSendCont ;
 PPar.         Proc  ::= Proc "|" Proc1 ;
 
 separator Proc "," ;
 
+-- Let declarations
+DeclImpl.    Decl ::= [Name] NameRemainder "<-" [Proc] ;
+
+LinearDeclImpl.    LinearDecl ::= Decl ;
+separator nonempty LinearDecl ";" ;
+
+ConcDeclImpl.    ConcDecl ::= Decl ;
+separator nonempty ConcDecl "&" ;
+
+EmptyDeclImpl.      Decls ::= ;
+LinearDeclsImpl.    Decls ::= ";" [LinearDecl] ;
+ConcDeclsImpl.      Decls ::= "&" [ConcDecl] ;
+
+-- Continuation for synchronous send
+
+EmptyCont.      SynchSendCont ::= "." ;
+NonEmptyCont.   SynchSendCont ::= ";" Proc1 ;
```

Comment:
I am trying to understand the bnfc grammar. 
Joe, do you have any recommend docs or examples for understand the whole design on bnfc?
I am just curious.

Reply:
I got https://bnfc.readthedocs.io/en/latest/lbnf.html#introduction. But if you got something, I'd love to take a look also.

Reply:
That's all I used. If there's a specific part that confuses you, I can try to help.